src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double MULTIPLICATION_INDEX = 1.5d;
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        extractedArray();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not exist in Array list");
+        }
+        extractedArray();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
+        validateIndex(index);
+        elementData[index] = value;
+    }
 
+    private void validateIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not exist in Array list");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        T removedValue = elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
```

Comment:
I think , we do not need use elementData[size] = null, because array moves to the left and the elementData[size]  will be elementData[size] = element [ index + 1] after arraycopy.

Reply:
если не сделать это действие, у тебя в конце будут 2 одинаковых значения и по факту размер массива не уменьшиться